WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -388,22 +388,28 @@ private extension ProductsViewController {
         ProductsTopBannerFactory.topBanner(isExpanded: isExpanded,
                                            isInAppFeedbackFeatureEnabled: isInAppFeedbackEnabled,
                                            expandedStateChangeHandler: { [weak self] in
-            self?.tableView.updateHeaderHeight()
+            self?.updateTableHeaderView()
         }, onGiveFeedbackButtonPressed: { [weak self] in
             self?.presentProductsFeedback()
         }, onDismissButtonPressed: { [weak self] in
             self?.dismissProductsBanner()
         }, onCompletion: { [weak self] topBannerView in
             self?.topBannerContainerView.updateSubview(topBannerView)
             self?.topBannerView = topBannerView
-            self?.tableView.updateHeaderHeight()
+            self?.updateTableHeaderView()
         })
     }
 
     func hideTopBannerView() {
         topBannerView?.removeFromSuperview()
         topBannerView = nil
-        tableView.tableHeaderView = nil
+        updateTableHeaderView()
+    }
+
+    /// Updates table header view with the correct spacing / edges depending if `topBannerContainerView` is empty or not.
+    ///
+    func updateTableHeaderView() {
+        topStackView.spacing = topBannerContainerView.subviews.isNotEmpty ? 8 : 0
```

Comment:
Where do these magic numbers come from? I guess this is from this declaration right? 

https://github.com/woocommerce/woocommerce-ios/blob/1c0bb1cfbb3d99a4f42efc7aaadab1ab1d9f8d2a/WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift#L42

If so, should we consider moving `8` to a constant (enum)?